### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: dependencyManagement/build.gradle.kts [hunk 1]
```diff
@@ -12,21 +12,22 @@ val dependencyVersions = hashMapOf<String, String>()
...
-  "org.junit:junit-bom:5.7.2",
+  "com.google.protobuf:protobuf-bom:3.17.3",
+  "com.linecorp.armeria:armeria-bom:1.9.2",
+  "com.squareup.okhttp3:okhttp-bom:4.9.1",
+  "io.grpc:grpc-bom:1.39.0",
+  "io.zipkin.brave:brave-bom:5.13.3",
   "io.zipkin.reporter2:zipkin-reporter-bom:2.16.3",
-  "com.squareup.okhttp3:okhttp-bom:4.9.0"
+  "org.junit:junit-bom:5.7.2",
+  "org.testcontainers:testcontainers-bom:1.16.0"
```
Comment (kubawach) [inline dependencyManagement/build.gradle.kts:24]: I see alphabetical sort... nice!

File: dependencyManagement/build.gradle.kts [hunk 2]
```diff
@@ -88,16 +84,16 @@ val DEPENDENCIES = listOf(
   "io.opentracing:opentracing-api:0.33.0",
   "io.zipkin.zipkin2:zipkin-junit:2.23.2",
   "junit:junit:4.13.2",
```
Comment (kubawach) [inline dependencyManagement/build.gradle.kts:86]: Out of curiosity - why do we have two junit versions ("org.junit:junit-bom:5.7.2" as BOM for the second one)?

**Review by kubawach — APPROVED** [review]
LGTM

File: dependencyManagement/build.gradle.kts (hunk 2 above)
Comment (jkwatson) [reply dependencyManagement/build.gradle.kts:86]: We have some test helpers for junit 4 as well as using junit 5 ourselves for everything.

**Review by jkwatson — APPROVED** [review]

**Pull request merged**